app.rb

```diff
@@ -0,0 +1,324 @@
+    
+    @books.each do |book|
+      puts "#{key} - #{book.title} by #{book.author}"
+      key += 1
+    end
+  end
+
+  def list_all_people
+    key = 1
+    puts
+    puts 'People'.upcase
+    puts
+    puts 'No people yet! Choose option 3 to add a person ' if @people.empty?
+    
+    @people.each do |person|
+      print "#{key} - [#{person.class.name} ID]: #{person.id} Name: #{person.name} "
+      print "Parent Permission: #{person.parent_permission}" if person.is_a?(Student)
+      print "Specialization: #{person.specialization}" if person.is_a?(Teacher)
+      puts
+      key += 1
+    end
+  end
+
+  def create_person
+    entry = nil
+    print 'Choose option 1 to create a Student or option 2 for a Teacher: '
+    
+    until [1, 2].include?(entry)
+      entry = gets.chomp.strip.to_i
+      puts
+      puts 'Choose option 1 for Student or option 2 for Teacher' unless [1, 2].include?(entry)
+    end
+
+    case entry
+    when 1
+      create_student
+    when 2
+      create_teacher
+    end
+  end
+
+  def create_student
+    print 'Enter Age: '
+    age = -1
+    while age <= 0
+      age = gets.chomp.to_i
+      print 'Enter valid age for student: ' if age <= 0
+    end
+
+    print 'Enter Name: '
+    name = gets.chomp.strip.capitalize
+
+    print 'Does student have parent permission? (Y/N): '
+    permission = gets.chomp.strip.upcase
+
+    case permission
+    when 'Y', 'YES'
+      permission = true
+    when 'N', 'NO'
+      permission = false
+    end
+    @people << Student.new(age: age, name: name, parent_permission: permission, classroom: nil)
+    puts
+    puts 'New student created successfuly!'
+    puts
+  end
+
+  def create_teacher
+    age = -1
+    print 'Enter Age: '
+    while age <= 0
+      age = gets.chomp.to_i
+      print 'Enter valid age for teacher: ' if age <= 0
+    end
+
+    print 'Enter name: '
+    name = gets.chomp.strip.capitalize
+
+    print 'Enter specialization: '
+    specialization = gets.chomp.strip.capitalize
+
+    @people << Teacher.new(age: age, name: name, specialization: specialization)
+    puts 'New teacher created successfuly!'
+  end
+
+  def create_book
+    print 'Enter title: '
+    title = gets.chomp.strip.capitalize
+
+    print 'Enter author: '
+    author = gets.chomp.strip.capitalize
+
+    @books << Book.new(title: title, author: author)
+    puts
+    puts 'New book was created successfully!'
+    puts
+  end
+
+  def create_rental
+    list_all_books
+
+    print 'Select the key of the book: '
+    
+    selected_book = gets.chomp.chomp.to_i
+
+    list_all_people
+
+    print 'Select the key of the person: '
+    selected_person = gets.chomp.chomp.to_i
+
+    print 'Select the date: (Year/Month/Day): '
+
+    date = gets.chomp.strip
+    book = @books[selected_book]
+    person = @people[selected_person]
+    new_rental = Rental.new(date: date, book: book, person: person)
+    @rentals.push(new_rental)
```

Comment:
The is something wrong with creating rentals on your app. I'll walk you through the problem.

- I created the following PEOPLE: a student (TOM) and a teacher (SALLY).

![image](https://user-images.githubusercontent.com/38088818/194118123-9632b281-1590-4337-b343-067ad4e463d1.png)

- Then created the following books.

![image](https://user-images.githubusercontent.com/38088818/194118339-c682fd38-f96d-4146-9cb0-b1efc2f3da4a.png)

- Then I created the first rental. As you can see below I selected option 1 for the book `The Fault in Our Stars` and optional 1 for the student `Tom` with ID `611`.

![image](https://user-images.githubusercontent.com/38088818/194118840-eed53def-658b-41b1-a30c-0d2eefeb045a.png)

Now look what happens when I check the rental I just created for `Tom` with ID `611`

![image](https://user-images.githubusercontent.com/38088818/194120510-354b4e18-722b-47b6-9257-b9f2b1ac00e7.png)

The rental listed for `Tom` is for `Sally` the teacher, and the book is not the one I selected either; its book 2 on the options `To Kill A Mocking Bird`.

Then I created a rental for the teacher (SALLY) option 2, with book 2 (TO KILL A MOCKING BIRD)  which is the rental that was created for TOM by mistake:

![image](https://user-images.githubusercontent.com/38088818/194121553-2e3e1060-3b16-4a60-9014-3e5403156875.png)

Now when I check the new rental there's an error:

![image](https://user-images.githubusercontent.com/38088818/194122292-d87e37a8-1262-458f-8c2d-cc0b1639881f.png)

- I hope my explanation makes sense. Kindly look into it and make sure that respective rentals are created based on the options selected for book and person.